xla/service/gpu/cudnn_norm_rewriter.h

```diff
@@ -0,0 +1,44 @@
+/* Copyright 2023 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef TENSORFLOW_COMPILER_XLA_SERVICE_GPU_CUDNN_NORM_REWRITER_H_
+#define TENSORFLOW_COMPILER_XLA_SERVICE_GPU_CUDNN_NORM_REWRITER_H_
+
+#include "absl/strings/string_view.h"
+#include "xla/hlo/ir/hlo_module.h"
+#include "xla/service/hlo_pass_interface.h"
+
+namespace xla {
+namespace gpu {
+
+class CudnnNormRewriter : public HloModulePass {
```

Comment:
The idea is to also use this pass for other cases, including the backward pass of the layer norm and other types of norms. I kept the comment more general.

Reply:
We should describe every type of pattern-match-and-replace that this pass does, since it's harder to determine this from the code. This should either be commented here in the header or on top of the function which does the rewrite, e.g. `MatchLayerNorm`. The current comment in `MatchLayerNorm` is good but it should also describe the definition of layer norm, especially considering how concise it is: `(x - Mean(x))/(Variance(x)**0.5) * scale + bias`

If you do comment in the function, you should add in the header comment: "For now, only the forward pass of layer norm is matched."